GVRf/Framework/jni/objects/mesh.cpp

```diff
@@ -180,12 +180,11 @@ void Mesh::getTransformedBoundingBoxInfo(glm::mat4 *Mat,
 void Mesh::generateVAO() {
 #if _GVRF_USE_GLES3_
 
-    if (vaoInitiliased_)
-        return;
+
     GLuint tmpID;
 
-    if (vao_dirty_) {
-         deleteVaos();
+    if (!vao_dirty_) {
+         return;
```

Comment:
We reuse the old VAO for the mesh in the “common” case. This function will regenerate a new VAO only if the mesh is dirty – if we changed the attributes that define the VAO.

From: Mihail [mailto:notifications@github.com]
Sent: Wednesday, October 07, 2015 8:23 AM
To: Samsung/GearVRf
Cc: Vittalbabu Srinivasan
Subject: Re: [GearVRf] Cleanup of mesh interface -- allow access only via const references. (#259)


In GVRf/Framework/jni/objects/mesh.cpp<https://github.com/Samsung/GearVRf/pull/259#discussion_r41403879>:

>      GLuint tmpID;

>

> -    if (vao_dirty_) {

> -         deleteVaos();

> +    if (!vao_dirty_) {

> +         return;

I am curious why the deleteVaos() call is no longer necessary in case of a "dirty" mesh?

—
Reply to this email directly or view it on GitHub<https://github.com/Samsung/GearVRf/pull/259/files#r41403879>.